vpr/src/analytical_place/analytical_placement_flow.cpp

```diff
@@ -58,6 +58,75 @@ static void print_ap_netlist_stats(const APNetlist& netlist) {
     VTR_LOG("\n");
 }
 
+/**
+ * @brief Passes the flat placement information to a provided partial placement.
+ *
+ *  @param flat_placement_info    The flat placement information to be read.
+ *  @param ap_netlist             The APNetlist that used to iterate over its blocks.
+ *  @param prepacker              The Prepacker to get molecule of blocks in the ap_netlist.
+ *  @param p_placement            The partial placement to be updated which is assumend
+ * to be generated on ap_netlist or have same blocks.
+ */
+static void convert_flat_to_partial_placement(const FlatPlacementInfo& flat_placement_info, const APNetlist& ap_netlist, const Prepacker& prepacker, PartialPlacement& p_placement){
+    for (APBlockId ap_blk_id : ap_netlist.blocks()) {
+        // Get the molecule that AP block represents
+        PackMoleculeId mol_id = ap_netlist.block_molecule(ap_blk_id);
+        const t_pack_molecule& mol = prepacker.get_molecule(mol_id);
+        // Get location of a valid atom in the molecule and verify that
+        // all atoms of the molecule share same placement information.
+        t_pl_loc atom_loc;
+        bool found_valid_atom = false;
+        for (AtomBlockId atom_blk_id: mol.atom_block_ids) {
+            if (!atom_blk_id.is_valid())
+                continue;
+            const t_pl_loc current_loc(flat_placement_info.blk_x_pos[atom_blk_id],
+                                       flat_placement_info.blk_y_pos[atom_blk_id],
+                                       flat_placement_info.blk_sub_tile[atom_blk_id],
+                                       flat_placement_info.blk_layer[atom_blk_id]);
+            if (found_valid_atom) {
+                VTR_ASSERT_MSG(current_loc == atom_loc, "Verify all atoms of a molecule provided in the flat placement share same location.\n");
```

Comment:
same location -> the same location

Reply:
I'd give more information, and say what the error is.
"Molecule " mol_id "contains atom %s (%d) with a location of " atom_loc "that conflicts with the location of other atoms in this molecule of " current_loc "."

Probably we should make this a VTR_ERROR (more user-friendly).